src/Application/Components/MapCanvas/MapCanvas.jsx

```diff
@@ -0,0 +1,58 @@
+import React from "react";
+import { fabric } from 'fabric-webpack';
+
+/**
+ * No Pin functionality yet.
+ */
+const MapCanvas = React.createClass({
+  propTypes: {
+    pins: React.PropTypes.object,
```

Comment:
objects as props are generally bad, since they're mutable and therefore can't be tested for shallow equality reliably.  this is the reason for immutablejs.